js/net.ts

```diff
@@ -0,0 +1,151 @@
+  ) {}
+
+  write(p: ArrayBufferView): Promise<number> {
+    return write(this.fd, p);
+  }
+
+  read(p: ArrayBufferView): Promise<ReadResult> {
+    return read(this.fd, p);
+  }
+
+  close(): void {
+    close(this.fd);
+  }
+}
+
+/** Listen announces on the local network address.
+ *
+ * The network must be "tcp", "tcp4", "tcp6", "unix" or "unixpacket".
+ *
+ * For TCP networks, if the host in the address parameter is empty or a literal
+ * unspecified IP address, Listen listens on all available unicast and anycast
+ * IP addresses of the local system. To only use IPv4, use network "tcp4". The
+ * address can use a host name, but this is not recommended, because it will
+ * create a listener for at most one of the host's IP addresses. If the port in
+ * the address parameter is empty or "0", as in "127.0.0.1:" or "[::1]:0", a
+ * port number is automatically chosen. The Addr method of Listener can be used
+ * to discover the chosen port.
+ *
+ * See dial() for a description of the network and address parameters.
+ */
+export function listen(network: Network, address: string): Listener {
+  const builder = new flatbuffers.Builder();
+  const network_ = builder.createString(network);
+  const address_ = builder.createString(address);
+  fbs.Listen.startListen(builder);
+  fbs.Listen.addNetwork(builder, network_);
+  fbs.Listen.addAddress(builder, address_);
+  const msg = fbs.Listen.endListen(builder);
+  const baseRes = dispatch.sendSync(builder, fbs.Any.Listen, msg);
+  assert(baseRes != null);
+  assert(fbs.Any.ListenRes === baseRes!.msgType());
+  const res = new fbs.ListenRes();
+  assert(baseRes!.msg(res) != null);
+  return new ListenerImpl(res.rid());
+}
+
+/** Dial connects to the address on the named network.
+ *
+ * Supported networks are only "tcp" currently.
+ * TODO: "tcp4" (IPv4-only), "tcp6" (IPv6-only), "udp", "udp4"
+ * (IPv4-only), "udp6" (IPv6-only), "ip", "ip4" (IPv4-only), "ip6" (IPv6-only),
+ * "unix", "unixgram" and "unixpacket".
+ *
+ * For TCP and UDP networks, the address has the form "host:port". The host must
+ * be a literal IP address, or a host name that can be resolved to IP addresses.
+ * The port must be a literal port number or a service name. If the host is a
+ * literal IPv6 address it must be enclosed in square brackets, as in
+ * "[2001:db8::1]:80" or "[fe80::1%zone]:80". The zone specifies the scope of
+ * the literal IPv6 address as defined in RFC 4007. The functions JoinHostPort
+ * and SplitHostPort manipulate a pair of host and port in this form. When using
+ * TCP, and the host resolves to multiple IP addresses, Dial will try each IP
+ * address in order until one succeeds.
+ *
+ * Examples:
+ *
+ *   dial("tcp", "golang.org:http")
+ *   dial("tcp", "192.0.2.1:http")
+ *   dial("tcp", "198.51.100.1:80")
+ *   dial("udp", "[2001:db8::1]:domain")
+ *   dial("udp", "[fe80::1%lo0]:53")
+ *   dial("tcp", ":80")
+ */
+export async function dial(network: Network, address: string): Promise<Conn> {
```

Comment:
Not sold on the `dial` name, but let's keep it for now.
However I would suggest to treat `connect` as a reserved method name, in case we change our mind.

Reply:
Added.